packages/server/src/handlers/BaseHandler.ts

```diff
@@ -151,9 +151,11 @@ export class BaseHandler extends EventEmitter {
         reject(err.name === 'AbortError' ? ERRORS.ABORTED : err)
       })
 
-      req.on('error', (err) => {
+      req.on('error', () => {
         if (!proxy.closed) {
-          proxy.destroy(err)
+          // we end the stream gracefully here so that we can upload the remaining bytes to the store
+          // as an incompletePart
+          proxy.end()
```

Comment:
A very small but nice change here.
When an upload request is canceled, Instead of calling `proxy.destroy()` we can call `proxy.end()` to gracefully close the passthrough stream, so that we can flush the bytes of a chunk as an incomplete part and upload it to S3.

This will result in starting exactly where we left off as opposed to starting from the last successful chunk uploaded.

Reply:
Nice!